src/components/CowCard/CowCard.test.js

```diff
@@ -0,0 +1,386 @@
+        color: cowColors.WHITE,
+        name: '',
+        baseWeight: 100,
+        gender: genders.FEMALE,
+      })
+
+      render(
+        <CowCard
+          {...{
+            ...baseProps,
+            cow: testCow,
+            cowInventory: [testCow, breedingCow1],
+            cowBreedingPen: {
+              cowId1: breedingCow1.id,
+              cowId2: null,
+              daysUntilBirth: -1,
+            },
+            money: 0,
+            handleCowSellClick: () => {},
+          }}
+        />
+      )
+
+      const button = screen
+        .getByText('Breed')
+        .closest('label')
+        .querySelector('[type=checkbox]')
+      expect(button).toHaveAttribute('disabled')
+    })
+
+    test('is disabled when pen is full', () => {
+      const testCow = generateCow({
+        color: cowColors.WHITE,
+        name: '',
+        baseWeight: 100,
+      })
+      const breedingCow1 = generateCow({
+        color: cowColors.WHITE,
+        name: '',
+        baseWeight: 100,
+        gender: genders.MALE,
+      })
+      const breedingCow2 = generateCow({
+        color: cowColors.WHITE,
+        name: '',
+        baseWeight: 100,
+        gender: genders.FEMALE,
+      })
+
+      render(
+        <CowCard
+          {...{
+            ...baseProps,
+            cow: testCow,
+            cowInventory: [testCow, breedingCow1, breedingCow2],
+            cowBreedingPen: {
+              cowId1: breedingCow1.id,
+              cowId2: breedingCow2.id,
+              daysUntilBirth: -1,
+            },
+            money: 0,
+            handleCowSellClick: () => {},
+          }}
+        />
+      )
+
+      const button = screen
+        .getByText('Breed')
+        .closest('label')
+        .querySelector('[type=checkbox]')
+      expect(button).toHaveAttribute('disabled')
+    })
+  })
+})
+
+describe('CowCardSubheader', () => {
+  let baseProps
+
+  beforeEach(() => {
+    jest.spyOn(Math, 'random').mockReturnValue(0)
+    baseProps = {
+      cow: generateCow({
+        color: cowColors.WHITE,
+        happiness: 0,
+        name: '',
+        baseWeight: 100,
+      }),
+      cowBreedingPen: { cowId1: null, cowId2: null, daysUntilBirth: -1 },
+      cowInventory: [],
+      cowValue: 1000,
+      huggingMachinesRemain: false,
+      isCowPurchased: false,
+    }
+  })
+
+  describe('happiness display', () => {
+    describe('cow is not purchased', () => {
+      test('renders no hearts', () => {
+        const { container } = render(<CowCardSubheader {...baseProps} />)
+
+        expect(container.querySelector('.heart')).toBeNull()
+      })
+    })
+
+    describe('cow is purchased', () => {
+      test('renders hearts', () => {
+        const { container } = render(
+          <CowCardSubheader {...{ ...baseProps, isCowPurchased: true }} />
+        )
+
+        expect(container.querySelectorAll('.heart')).toHaveLength(10)
+      })
+
+      test('renders full hearts that match cow happiness', () => {
```

Comment:
i would break this test up into a couple pieces. have expects -> render -> expects can cause for some whack a mole when things break. i would at least make two pieces here, since you could simply insert another `test` line after the first two expects, but i would also reconsider if they actually fit together. i see a "it renders all 10 hearts regardless of current happiness level", and also a "it renders full hearts to represent current happiness". i almost suggested combining the expects when i saw the last 2 because it seems like expecting .heart.is-full to have length of 10 covers both "are there 10 hearts and are they full?" but i think your intention was different. so creating the separate `test`'s will better express this.

to take it a step further, you could go with a loop that increments from 1-10 and then just write 2 `test`'s inside, one that makes sure there's always 10 hearts, and one that makes sure there are N full hearts.

Reply:
> i would break this test up into a couple pieces. have expects -> render -> expects can cause for some whack a mole when things break. i would at least make two pieces here, since you could simply insert another test line after the first two expects

This is a great idea! Done in https://github.com/jeremyckahn/farmhand/commit/5f4fe0c5206fbbba36dd00ec4b978f8bde0d6e69. I also removed the two `expect(container.querySelectorAll('.heart')).toHaveLength(10)`s in this test as they seemed redundant given the expectation in the `'renders hearts'` test above.

> to take it a step further, you could go with a loop that increments from 1-10 and then just write 2 `test`'s inside, one that makes sure there's always 10 hearts, and one that makes sure there are N full hearts.

I was thinking about that too, but I generally try to avoid dynamic tests as they introduce their own complexity. They can be an appropriate solution, but in this case I don't think the value that dynamic tests would give us here would justify the complexity cost.